File: BMSTU-2/OOP/test3.cpp
#pragma GCC optimize("unroll-loops", "O3")
#include "iostream"
using std::cout, std::cin;
int main()
{
    std::ios::sync_with_stdio(0);
    cin.tie(0);
    long long mx, mn, k;
    cin >> mx >> mn;
    if (mx < 0 and mn < 0)
    {
        mx = -1 * mx;
        mn = -1 * mn;
    }
    if (mx < mn)
    {
        mx += mn;
        mn = mx - mn;
        mx -= mn;
    };
    if (abs(mx - mn) < 1)
    {
        cout << (mx % 3 == 2);
        return 0;
    }
    if (mn < 0)
    {
        cout << (abs(mn) - 2) / 3 + 1 + (mx - 2) / 3 + 1;
    }
    else
    {
        cout << (mx - mn - (2 - abs(mn % 3))) / 3 + 1;
    }
    return 0;
}

File: BMSTU-2/OOP/test3.py
a, b = map(int, input().split())
if a < b:
    a, b = b, a
if (abs(a)-abs(b) < 2):
    print(0 if a % 3 != 2 else 1)
else:
    while (a % 3 != 2):
        a -= 1
    if a > 0 and b > 0:
        k = a-b
        print(abs(k//3+1))
    elif a > 0 and b < 0:
        b = abs(b)
        while (b % 3 != 2):
            b -= 1
        print(abs(a//3+1)+abs(b//3+1))
    else:
        k = abs(a-b)
        print(abs(k//3+1))
